src/main/scala/beam/agentsim/agents/vehicles/BeamVehicle.scala

```diff
@@ -54,6 +54,7 @@ class BeamVehicle(
 
   val rand: Random = new Random(randomSeed)
 
+  @volatile
   var spaceTime: SpaceTime = _
```

Comment:
`volatile` because there are places outside of `BeamVehicle` which set `spaceTime`